core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -90,30 +93,239 @@
   /** Whether to cache blocks in this file into Alluxio. */
   private final boolean mShouldCache;
 
-  // The following 3 fields must be kept in sync. They are only updated in updateStreams together.
+  // The following variables should be kept in sync in updateStreams.
+  /** Position states. */
+  protected PositionState mPositionState;
+  /** Current block ID. */
+  private long mCurrentBlockId = UNINITIALIZED_BLOCK_ID;
+  /** Current block's start position. */
+  private long mCurrentBlockStartPos = UNINITIALIZED_BLOCK_POS;
```

Comment:
can we use `mPositionState .getPositionState.getBlockStartPos()` to replace this field? The reason is it seems we put all pos related fields into the state object. If that is the convention, we just always get from there rather than having another copy of a sub-field.

Reply:
same argument applies to `mCurrentBlockEndPos`

Reply:
Yeah, I feel like there should be no manual "syncing" of state in this outer class. All the state should be maintained by the `PositionState`.

Reply:
the block in `PositionState` is the block containing the current position, for example, if `mPositionState.setPos(pos + mBlockSize)`, then the block in `PositionState` is updated to the next block, but the `mCurrentBlockInStream`, `mCurrentBlockStartPos`, etc are not updated yet, until `updateStreams` is called.

Reply:
We can move the `mCurrentBlockInStream` and `mCurrentCacheStream` into `PositionState`, and every time the position changes, we update both streams to keep in sync.

But for `seek` or `skip`, the streams do not need to be updated unless data needs to be read, always updating streams when position changes might bring unnecessary cost.

Reply:
But, isn't `mCurrentBlockStartPos` only related to the position? I don't think streams belong in the state, the start and end position seem like position state, and they can all be kept in sync correctly.